### ISSUE TITLE
Table Of Content - no indentation

### ISSUE THREAD

**Issue by omribeni** [xceedsoftware/DocX#227]
Hi, the table of content does not add proper indentation. Is there a way to change the table of content's format, because when i change it manually to "Simple" rather then "From template" the indentation appears.

Has anybody faced this issue? please let me know.

thanks.

**Comment by omribeni** [comment]
![capture](https://user-images.githubusercontent.com/21240978/41524986-2817d3d4-72e8-11e8-898a-8212a98fac12.PNG)

For some reason using your Table of Content even changes the word picture to not have indentation
**the  "Format: from Template" may be the problem but there is no way i saw to change it.

please help :)

**Comment by XceedBoucherS** [comment]
Hi,
Thank you for the screenshot.

But, how exactly do you create the Table of Content.
You can have a look at our sample in "Samples/TableOfContent/TableOfContentSample.cs".
Or you can paste a simple sample showing the issue.
Thank you.

**Comment by Wayne-Mather** [comment]
I have the same issue.

```
using(var doc = Xceed.Words.NET.DocX.Create(fullPathToFileName))
{
     doc.InsertTableOfContents("Table of Contents",
                TableOfContentsSwitches.O // use paragraphs with built in heading styles
                | TableOfContentsSwitches.U // build the toc by using paragraph outline level
                | TableOfContentsSwitches.Z // hide tba leader in web layout
                | TableOfContentsSwitches.H // use hyperlinks
                | TableOfContentsSwitches.W // preserve tabs
                | TableOfContentsSwitches.X // preserve new line characters
                );
            doc.InsertSectionPageBreak();

      var para = doc.InsertParagraph();
      para.Bold();
      para.InsertText("First");
      para.StyleName = "Heading1";

     para = doc.InsertParagraph();
     para.Bold();
     para.InsertText("Second");
     para.StyleName = "Heading2";

     para = doc.InsertParagraph();
     para.Bold();
     para.InsertText("Third");
     para.StyleName = "Heading1";

     para = doc.InsertParagraph();
     para.Bold();
     para.InsertText("Fourth");
     para.StyleName = "Heading2";

     doc.Save();
}

```

This is just an example of how I am processing my documents. However, if you generate and open in Word 2016, you will see that the TOC does not have any indentation. 

The current workaround for myself is to go to the Table of Contents under references in the Ribbon, under General->Formats change the drop-down to Formal and click Ok.

Hope this helps anyone else.

**Comment by Scantraxx123** [comment]
Did somebody an automatic solution? I used @Wayne-Mather method and it worked, but it would be better if I could do it in code.

**Comment by Wayne-Mather** [comment]
@Scantraxx123 - Judging by the feedback from the new authors and the repository activity, I don't think that they are really active or engage the community for the open source library. 

It's a shame because @PrzemyslawKlys created a great library and supported it before it was sold (I pressume) to Xceed.

I have decided to move away from this software when I can but have not found an alternative at this time. My current thoughts are to actually generate LaTex so I can then generate PDF, HTML, RTF.

**Comment by PrzemyslawKlys** [comment]
@Wayne-Mather Sooner or later this will get fixed. There is a promise that the free version is few versions behind. I assume that if it gets fixed in 1.7 it will get to free version in 1-2 years. Unfortunately, it's business and they have to make money somehow on the pay version. Unfortunately, some features that I really would like to see are supposed to be in pay version only (Core support). It's not an ideal scenario but let's not forget a few years back it's not like people added new features or fixed bugs on a daily basis :-) Anything that requires this commitment has to cost money. I've actually moved into PowerShell and things I do best (system admin stuff). I do have PowerShell module based on DocX called PSWriteWord and I'm using a free version of Xceed just like everyone else. And those that need features ASAP or do it for their business can spend those 530 euro and get the most up to date version there is.

**Comment by Scantraxx123** [comment]
Sad to hear that, when I first found this libary I really liked it, but now its a little bit frustrating to always here, thats its only available in the pay version. ..
I would be very happy if someone found a good replacement for this, could give me the name if it :)

**Comment by PrzemyslawKlys** [comment]
There is no good replacement and I doubt there will be. It takes a lot of effort to build this one. You have to wait for new features and fixes but at least you get them ;-) For free that is. You know you can still fork it and some people do and work your way on the new version. But I doubt it will get traction since it needs a dedicated team to add new features to it.

**Comment by PrzemyslawKlys** [comment]
@Wayne-Mather I agree. They may not seem very supportive here and only occasionally respond to issues (I guess once in 2 months or so, or when a new release is coming soon) but they still do. They just may not have enough resources to run full throttle community support. And since most fixes go to newest, pay version first, there's not much they could do I guess right? Other than asking for what is needed all they can do is update issue and say, fixed in 1.6 as I've seen in other threads. But what does that give you? Not much because it's pay version only and we don't know if it gets added back to free one. And if you really want you can go for the pay version. 

1.6 pay version was just released and I see it has support for CORE. It's too bad there was a decision to keep it in pay version only. Hope they change their minds about it. In a few years without CORE on the free version, this product (free version of it) may be dead. 

My guess is that this particular issue you're having will get fixed in 1.7 or 1.8. But you will get it in free version in like 2 years. Your best bet, for now, is to either wait for the pay version fix and then buy it or look for another product (there are few on the market). It doesn't seem like anyone wants to build another free product like this. 

To summarize, having a community and free product support is hard to maintain for a small company (and I don't think Xceed is huge.

**Comment by XceedBoucherS** [comment]
Hi,
You can try with the latest version v1.6 version available for free for 45 days here : https://xceed.com/xceed-words-for-net/.
The TOC indentations should already be fixed in that version.

As @PrzemyslawKlys said, Xceed is still maintaining the community edition of this product and will continue to do so. Of course, users of the commercial edition will receive faster support and have more features, but once in a while the users of the community edition also get support. Xceed is happy to get the users feedbacks in order to prioritize the next features/fix to work on. 

This product is only one of the products Xceed is working on and they put great effort in it in order to have a powerful tools to help developers around the world work with docx documents. If you do not want to pay for a commercial version, that is no problem, you can always try free trial for 45 days or continue with the community edition, which will continue to be about 2 or 3 versions behind the commercial edition.
